### PR TITLE
bpo-30553: Add status code 421 (Misdirected Req.) to http.HTTPStatus

### PR DESCRIPTION
**Pull request by vmsp**
https://bugs.python.org/issue30553

### CODE REVIEW

File: Lib/http/__init__.py [hunk 1]
```diff
@@ -98,6 +99,8 @@ def __new__(cls, value, phrase, description=''):
         'Cannot satisfy request range')
     EXPECTATION_FAILED = (417, 'Expectation Failed',
         'Expect condition could not be satisfied')
+    MISDIRECTED_REQUEST = (421, 'Misdirected Request',
+        'Server can not produce a response')
```
Comment (JelleZijlstra) [inline Lib/http/__init__.py]: "cannot" should be one word (unless this is specified otherwise in the RFC)

File: Doc/library/http.rst [hunk 2]
```diff
@@ -98,6 +98,7 @@ Code    Enum Name                           Details
 ``415`` ``UNSUPPORTED_MEDIA_TYPE``          HTTP/1.1 :rfc:`7231`, Section 6.5.13
 ``416`` ``REQUEST_RANGE_NOT_SATISFIABLE``   HTTP/1.1 Range Requests :rfc:`7233`, Section 4.4
 ``417`` ``EXPECTATION_FAILED``              HTTP/1.1 :rfc:`7231`, Section 6.5.14
+``421`` ``MISDIRECTED_REQUEST``             HTTP/2 :rfc:`7540`, Section 9.1.2
```
Comment (berkerpeksag) [inline Doc/library/http.rst:101]: Could you add

```rst
.. versionchanged:: 3.7
   Added ``421`` status code.
```

to the end of the file? You can adjust the "Added 421 status code." part if you want -- I just used it as an example :)

File: Misc/NEWS.d/next/Library/2017-07-05-14-48-26.bpo-30553.Oupsxo.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Add HTTP/2 status code 421 (Misdirected Request) to http.HTTPStatus.
```
Comment (berkerpeksag) [inline Misc/NEWS.d/next/Library/2017-07-05-14-48-26.bpo-30553.Oupsxo.rst]: Please add "Patch by Vitor Pereira.".

Comment (berkerpeksag) [inline Misc/NEWS.d/next/Library/2017-07-05-14-48-26.bpo-30553.Oupsxo.rst]: http.HTTPStatus -> :class:\`http.HTTPStatus\`

**Review by berkerpeksag — CHANGES_REQUESTED** [review]

File: Misc/NEWS.d/next/Library/2017-07-05-14-48-26.bpo-30553.Oupsxo.rst (hunk 3 above)
Comment (vmsp) [reply Misc/NEWS.d/next/Library/2017-07-05-14-48-26.bpo-30553.Oupsxo.rst]: Is this new? I sent a patch a couple of months ago and wasn't asked to do this.

**Comment by vmsp** [comment]
I have made the requested changes; please review again

File: Misc/NEWS.d/next/Library/2017-07-05-14-48-26.bpo-30553.Oupsxo.rst (hunk 3 above)
Comment (berkerpeksag) [reply Misc/NEWS.d/next/Library/2017-07-05-14-48-26.bpo-30553.Oupsxo.rst]: No, it's not new. We always try to add contributor's name in the NEWS file. Our GitHub workflow is still new so it's normal to forget this step (or sometimes we skip it intentionally if the patch is trivial)

**Review by berkerpeksag — APPROVED** [review]
LGTM, thank you!. I just made a small markup tweak and will merge it once Travis CI is happy.

**Comment by vmsp** [comment]
Cheers!

**Pull request merged**